app/models/lockstep/magic_link.rb

```diff
@@ -0,0 +1,10 @@
+class Lockstep::MagicLink < Lockstep::ApiRecord
+  self.model_name_uri = "v1/useraccounts/magic-links"
+  self.id_ref = "magicLinkId"
+  self.query_path = ""
+
+  def self.generate_magic_link(email_id, expiry, app_id, user_role)
```

Comment:
I like that you have a separate method for this instead of using the regular create. This makes sense.

The name magic link exists in the class name already. It feels like it is repeating to have the method as generate_magic_link. It might be better to have it as just generate, so the dev calls `Lockstep::MagicLink.generate`